angular/demo/daisyui/src/app/samples/modal/modal.component.ts

```diff
@@ -0,0 +1,58 @@
+import type {ModalWidget} from '@agnos-ui/angular-headless';
+import {BaseWidgetDirective, UseDirective, auBooleanAttribute, callWidgetFactory, createModal} from '@agnos-ui/angular-headless';
+import {ChangeDetectionStrategy, Component, type ElementRef, effect, input, viewChild, model} from '@angular/core';
+
+@Component({
+	selector: 'app-modal',
+	imports: [UseDirective],
+	template: `
+		<dialog class="modal modal-bottom sm:modal-middle" (close)="api.close()" #dialog>
+			<div class="modal-box">
+				<ng-content />
+				@if (state().closeButton) {
+					<form method="dialog">
+						<button class="btn btn-sm btn-circle btn-ghost absolute right-2 top-2" [auUse]="widget.directives.closeButtonDirective">✕</button>
+					</form>
+				}
+			</div>
+			@if (closeOnOutsideClick()) {
+				<form method="dialog" class="modal-backdrop">
+					<button (click)="api.close()">Close</button>
+				</form>
+			}
+		</dialog>
+	`,
+	standalone: true,
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ModalComponent extends BaseWidgetDirective<ModalWidget> {
+	readonly closeOnOutsideClick = input(undefined, {transform: auBooleanAttribute});
+	readonly closeButton = input(undefined, {transform: auBooleanAttribute});
+	readonly ariaCloseButtonLabel = input<string>();
+	readonly visible = model(false);
+
+	readonly dialog = viewChild.required<ElementRef<HTMLDialogElement>>('dialog');
+
+	constructor() {
+		super();
+		effect(() => {
+			if (this.state().visible) {
+				this.dialog().nativeElement.showModal();
```

Comment:
As discussed, I think it would be good to move this logic to a directive to put in the core.